src/arch.rs

```diff
@@ -24,7 +27,17 @@ mod x86_64;
 pub trait AbiArgument {
     /// Serialize the argument into the buffer. This method should keep track of arch-dependent
     /// stuff like register vs stack allocation.
-    fn to_bytes(&self, buffer: &mut Vec<u8>) -> Result<()>;
+    fn to_bytes(
+        &self,
+        buffer: &mut Vec<u8>,
+        find_dict_overrides: impl Copy
+            + Fn(
+                &ConcreteTypeId,
+            ) -> (
+                Option<extern "C" fn(*mut c_void, *mut c_void)>,
+                Option<extern "C" fn(*mut c_void)>,
+            ),
```

Comment:
Could you add a comment explaining is the argument necessary?

Also, maybe we could define a type alias and use it everywhere else, as it's quite verbose. Do you think it will make it more readable?

Reply:
We can't make type aliases for impl types in stable Rust yet (it's unstable).

I can add a comment, but I'm not sure where I should add it. I'll add it where we actually use it (in `Value::to_ptr`).

Reply:
Aah, it's a pity.

Yes, you could add it to Value::to_ptr, and maybe short a mention to it in the trait definition, pointing developers in the Value::to_ptr documentation